### ISSUE TITLE
Unexpected character '#'

### ISSUE THREAD

**Issue by coreybutler** [eslint/espree#283]
I know it's probably not the intended use, but I was trying to parse a node file with this and received this error. The node script is designed to be a command line tool, and espree is complaining about the shebang (`#!/usr/bin/env node`). 

**Stack Trace:**

```sh
SyntaxError: Unexpected character '#'
    at Parser.instance.raise.instance.raiseRecoverable (node_modules/espree/espree.js:390:19)
    at Parser.pp.getTokenFromCode (node_modules/acorn/dist/acorn.js:2831:8)
    at Parser.pp.readToken (node_modules/acorn/dist/acorn.js:2526:15)
    at Parser.pp.nextToken (node_modules/acorn/dist/acorn.js:2518:71)
    at Parser.parse (node_modules/acorn/dist/acorn.js:1631:10)
    at Object.parse (node_modules/acorn/dist/acorn.js:885:44)
    at Object.parse (node_modules/espree/espree.js:715:21)
```

This is pretty easily fixed by stripping the first line out, but it requires an extra pre-processing step. It'd be nice to either throw a warning or ignore these (or both) without crashing the process.

**Comment by coreybutler** [comment]
I didn't realize acorn had a property to do this. So, if anyone else comes across this, just add `allowHashBang: true` to the parsing properties.

I'm still going to leave this open, but as an enhancement request instead of a bug. A warning like "Did you forget to set allowHashBang: true?" or something along those lines would be a time saver.

**Comment by nzakas** [comment]
It's invalid JavaScript, so you have to strip that comment before parsing (that's what ESLint does).

**Comment by coreybutler** [comment]
@nzakas - yeah, that's exactly what I stated. It would still be nice to not have to have a pre-process to do that.

**Comment by walshie4** [comment]
FWIW it looks like acorn supports skipping this.  You'll just need to make sure this option gets passed in - https://github.com/acornjs/acorn/blob/dfcdbe4014cee1996129a8275bae28380cccd20c/acorn/src/state.js#L78

**Comment by platinumazure** [comment]
@eslint/eslint-team What do you think about [this comment](https://github.com/eslint/espree/issues/283#issuecomment-481016308)? On the one hand it doesn't seem critical to implement, but on the other hand, we could dogfood it in ESLint and get rid of some special logic.

**Comment by not-an-aardvark** [comment]
It seems like we probably need the special logic anyway for parsers that don't support the option.

**Comment by silverwind** [comment]
There is a [stage 3 proposal](https://github.com/tc39/proposal-hashbang) to support hashbangs in the language, should this issue be re-opened?

**Comment by silverwind** [comment]
Proposal reached stage 4, so it will eventually be in acorn (https://github.com/acornjs/acorn/issues/1134).

**Comment by mdjermanovic** [comment]
This is now supported in espree v9.4.0 with `ecmaVersion: 2023`.
